src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -166,20 +174,32 @@ AsyncReplResult<> RaftReplDev::replace_member(replica_id_t member_out_uuid, repl
             // entry and call exit_group() and leave().
             return m_msg_mgr.rem_member(m_group_id, member_out_uuid)
                 .via(&folly::InlineExecutor::instance())
-                .thenValue([this, member_out](auto&& e) -> AsyncReplResult<> {
+                .thenValue([this, member_out, commit_quorum](auto&& e) -> AsyncReplResult<> {
                     if (e.hasError()) {
                         // Its ok to retry this request as the request
                         // of replace member is idempotent.
                         RD_LOGE("Replace member failed to remove member : {}", e.error());
+                        reset_quorum_size(0);
                         return make_async_error<>(ReplServiceError::RETRY_REQUEST);
                     } else {
                         LOGINFO("Replace member removed member={} from group_id={}", member_out, group_id_str());
                     }
+
+                    // Revert the quorum size back to 0.
+                    reset_quorum_size(0);
                     return make_async_success<>();
                 });
         });
 }
 
+void RaftReplDev::reset_quorum_size(uint32_t commit_quorum) {
+    RD_LOGI("Reset raft quorum size={}", commit_quorum);
+    nuraft::raft_params params = raft_server()->get_current_params();
+    params.with_custom_commit_quorum_size(commit_quorum);
+    params.with_custom_election_quorum_size(commit_quorum);
```

Comment:
I am not sure it is right to set election_quorum_size = 1 as once the member_in added, it is possible they both elected themselves as leader.

But on the other side,  w/o this value,  if the last member alive is not the leader ,  can we proceed?